pysat/_constellation.py

```diff
@@ -167,6 +172,42 @@ def __init__(self, platforms=None, names=None, tags=None, inst_ids=None,
                 if len(log_msg) > 0:
                     logger.warning("; ".join(log_msg))
 
+                # Set the Constellation attributes
+                self.platforms = added_platforms
+                self.names = added_names
+                self.tags = added_tags
+                self.inst_ids = added_inst_ids
+        else:
+            # Set the Constellation attributes to be empty lists
+            self.platforms = []
+            self.names = []
+            self.tags = []
+            self.inst_ids = []
+
+        inst_len = len(self.instruments)
+
+        # Include Instruments from the constellation module, if it exists
+        if const_module is not None:
+            if not hasattr(const_module, 'instruments'):
+                raise AttributeError("missing required attribute 'instruments'")
+            self.instruments.extend(const_module.instruments)
+
+        # Add any Instruments provided in the list
+        if instruments is not None:
+            test_instruments = np.asarray(instruments)
```

Comment:
Potential place for `pysat.utils.listify` though you lose the check on input.

Reply:
Yeah, I think the check on input is important.  There should probably be an additional check that the list-elements are pysat.Instrument objects, but I think I should make a separate issue for that.